examples/test-LwIP-netconn/scpi_server.h

```diff
@@ -28,14 +28,29 @@
 
 #ifndef _SCPI_SERVER_H_
 #define _SCPI_SERVER_H_
+#ifdef __cplusplus
+extern "C" {
+#endif
 
+#define SCPI_DEVICE_PORT  5025 // scpi-raw standard port
+#define SCPI_CONTROL_PORT 5026 // libscpi control port (not part of the standard)
 
 #include <stdint.h>
+#include "api.h"
+#include "scpi/types.h"
 
 void scpi_server_init(void);
 
 void SCPI_AddError(int16_t err);
 void SCPI_RequestControl(void);
 
+// optional event handlers
+void SCPI_Event_DeviceConnected(struct netconn * conn);
```

Comment:
It would be great to have also `scpi_t * context` as the first parameter for all events.

Reply:
realized in follow-up commit. It seems to me that &scpi_context would need to be passed n scpi_server_thread()  to toprocessIoListen(), processSrqIoListen(), processIo() and processSrqIo(). This requires the respective signatures to be extended. If that's what you implied, I can add it in another commit.

Reply:
see a565c895080f17deda47062e00f76a27040e6c17